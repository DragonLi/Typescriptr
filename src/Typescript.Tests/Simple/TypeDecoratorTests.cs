using Assent;
using Typescriptr;
using Typescriptr.Formatters;
using Xunit;

namespace Typescript.Tests.Simple
{
    public class TypeDecoratorTests
    {
        class SimpleTypesOnly
        {
            public int IntType { get; set; }
            public long LongType { get; set; }
            public string StringType { get; set; }
            public decimal DecimalType { get; set; }
        }
        class TypeWithNestedType
        {
            public SimpleTypesOnly SimpleType { get; set; }
        }
        class TypeWithEnum
        {
            public EnumType AnEnum { get; set; }
        }
        public enum EnumType
        {
            FirstEnum,
            SecondEnum,
            ThirdEnum
        }

        [Fact]
        public void Generator_CommentDecoration_GenerateSuccessfully()
        {
            var generator = TypeScriptGenerator.CreateDefault().WithTypeDecorator(CommentBlockDecorator.Decorate);
            var generated = generator.Generate(new[]
            {
                typeof(SimpleTypesOnly),
                typeof(TypeWithNestedType),
                typeof(TypeWithEnum)
            });
            
            this.Assent(generated.JoinTypesAndEnums());
        }
    }
}