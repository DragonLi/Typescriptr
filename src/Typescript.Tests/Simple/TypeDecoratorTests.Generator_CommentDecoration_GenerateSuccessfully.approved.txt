declare namespace Api {
  /*
   * NB: Do not edit this class manually
   * Source: Typescript.Tests.Simple.TypeDecoratorTests+TypeWithEnum
   * Generated using Typescriptr https://github.com/gkinsman/Typescriptr
   */
  interface TypeWithEnum {
    anEnum: 'FirstEnum' | 'SecondEnum' | 'ThirdEnum';
  }
  /*
   * NB: Do not edit this class manually
   * Source: Typescript.Tests.Simple.TypeDecoratorTests+TypeWithNestedType
   * Generated using Typescriptr https://github.com/gkinsman/Typescriptr
   */
  interface TypeWithNestedType {
    simpleType: SimpleTypesOnly;
  }
  /*
   * NB: Do not edit this class manually
   * Source: Typescript.Tests.Simple.TypeDecoratorTests+SimpleTypesOnly
   * Generated using Typescriptr https://github.com/gkinsman/Typescriptr
   */
  interface SimpleTypesOnly {
    intType: number;
    longType: number;
    stringType: string;
    decimalType: number;
  }
}

---
/*
 * NB: Do not edit this class manually
 * Source: Typescript.Tests.Simple.TypeDecoratorTests+EnumType
 * Generated using Typescriptr https://github.com/gkinsman/Typescriptr
 */
enum EnumType {
  FirstEnum = 'FirstEnum',
  SecondEnum = 'SecondEnum',
  ThirdEnum = 'ThirdEnum',
}
